superdesk/publish/http_push.py

```diff
@@ -0,0 +1,156 @@
+
+        @param url: the resource url
+        @param formatted_item: string containing the item in the required format
+        @return: HTTPResponse object
+        '''
+        request_client = self._request_client(url)
+        request_client.request('POST', url, formatted_item, self.headers)
+        response = request_client.getresponse()
+        if response.status == 409:
+            raise IdConflict()
+        if response.status != 201:
+            raise Exception('Error pushing item %s' % response.reason)
+        return response
+
+    def _headers(self, *custom_headers):
+        '''Return headers for an HTTP request
+
+        @param custom_headers: tuples of header-value pairs
+        '''
+        updated_headers = copy(self.headers)
+        updated_headers.update(custom_headers)
+        return updated_headers
+
+    def _patch(self, item_url, _id, formatted_item, etag):
+        '''Send a patch request to content API to update a given item
+        Raises ItemNotFound if the item identified by _id did not exist
+        Raises Exception on other errors
+
+        @param item_url: the format of the item url
+        @param _id: the item identifier
+        @param formatted_item: string containing the item in the required format
+        @param etag: the etag of the existing item
+        @return: HTTPResponse object
+        '''
+        url = item_url.format(_id)
+        headers = self._headers(self.headers, ('If-Match', etag))
+        request_client = self._request_client(url)
+        request_client.request('PATCH', url, formatted_item, headers)
+        response = request_client.getresponse()
+        if response.status == 404:
+            raise ItemNotFound()
+        if response.status < 200 or response.status >= 300:
+            raise Exception('Error pushing item %s' % response.reason)
+        return response
+
+    def _request_client(self, url):
+        '''Return a HTTP request object that corresponds to the given url
+        Raises PublishHTTPPushError on invalid url
+
+        @param url: The URL for which to return the HTTP request client object
+        '''
+        url_parts = urlparse(url)
+        if url_parts.scheme.lower() == 'http':
+            return HTTPConnection(url_parts.netloc)
+        elif url_parts.scheme.lower() == 'https':
+            return HTTPSConnection(url_parts.netloc)
+        else:
+            raise PublishHTTPPushError.httpPushError(InvalidURL('Invalid HTTP URL %s' % url), config)
+
+    def _copy_published_media_files(self, item):
+        '''Copy the media files for the given item to the publish_items endpoint
+
+        @param item: the item object
+        '''
+        for k, v in item.get('renditions', {}).items():
+            if k == 'original_source':
+                # if the source is AAP Multimedia then don't copy
+                continue
+
+            del v['href']
+            if not app.media.exists(v['media'], resource='publish_items'):
```

Comment:
i'm not sure if we should post binaries, imo would be better to send urls where those can be fetched